docs/samples/specification/multiapi/generated/azure/multiapi/sample/v2/operations/_operation_group_two_operations.py

```diff
@@ -34,27 +35,28 @@ def build_test_four_request(
     **kwargs  # type: Any
 ):
     # type: (...) -> HttpRequest
-    api_version = kwargs.pop('api_version', "2.0.0")  # type: str
+    _headers = case_insensitive_dict(kwargs.pop("headers", {}) or {})
+    _params = case_insensitive_dict(kwargs.pop("params", {}) or {})
+
+    api_version = kwargs.pop('api_version', _params.pop('api-version', "2.0.0"))  # type: str
     parameter_one = kwargs.pop('parameter_one')  # type: bool
+    accept = _headers.pop('Accept', "application/json")
 
-    accept = "application/json"
     # Construct URL
     _url = kwargs.pop("template_url", "/multiapi/two/testFourEndpoint")
 
     # Construct parameters
-    _query_parameters = kwargs.pop("params", {})  # type: Dict[str, Any]
-    _query_parameters['parameterOne'] = _SERIALIZER.query("parameter_one", parameter_one, 'bool')
-    _query_parameters['api-version'] = _SERIALIZER.query("api_version", api_version, 'str')
+    _params['parameterOne'] = _SERIALIZER.query("parameter_one", parameter_one, 'bool')
```

Comment:
Hi @iscai-msft , I think this can be a problem for some build_xx_request functions, so I added changes at line 42 of this file (as generated result) in commit https://github.com/Azure/autorest.python/pull/1183/commits/8db1a0855ba41c4658395db5c79b9284309bf4fb

Add I added some test case for this scenario: https://github.com/Azure/autorest.python/pull/1183/commits/20bfab28abe591795c46eef1e8d60e049fd12044

Reply:
hey @changlong-liu sorry I should've thought more before I added this comment. Even though these required parameters are kwargs in the builders (and in some version tolerant operations), they are still required and there will be an error if users don't input them, so I think the reversal I did still stands: we don't need to check `headers` or `params` because users will be required to input and we want to keep raising the error if users don't put them in. What do you think?
<img width="563" alt="Screen Shot 2022-03-22 at 11 13 49 AM" src="https://user-images.githubusercontent.com/43154838/159515195-12875dc7-2c03-4214-8620-ebddccafeaae.png">

Reply:
Thanks for the talk in meeting about this.  Keeping current logic is the best way to ensure same behavior across different style of build_xx_request signatures.
So I reverted changes in https://github.com/Azure/autorest.python/commit/8db1a0855ba41c4658395db5c79b9284309bf4fb and https://github.com/Azure/autorest.python/commit/20bfab28abe591795c46eef1e8d60e049fd12044